File: test_login.py
from playwright.sync_api import sync_playwright
import time

def test_login():
    with sync_playwright() as p:
        browser = p.chromium.launch(headless=True)
        page = browser.new_page()
        
        try:
            print("Opening application...")
            page.goto("http://localhost:5173")
            
            # Wait for page to load
            page.wait_for_timeout(2000)
            
            # Take screenshot of current state
            page.screenshot(path="screenshot_complete_login.png")
            print("Screenshot saved as screenshot_complete_login.png")
            
            # Try to fill login form if visible
            try:
                # Wait for username field
                page.wait_for_selector('input[type="text"]', timeout=3000)
                
                # Fill credentials
                print("Filling login form...")
                page.fill('input[type="text"]', 'admin')
                page.fill('input[type="password"]', 'password')
                
                # Click login button
                page.click('button[type="submit"]')
                
                # Wait for response
                page.wait_for_timeout(3000)
                
                # Take screenshot after login
                page.screenshot(path="screenshot_complete_dashboard.png")
                print("Screenshot saved as screenshot_complete_dashboard.png")
                
                # Check current URL
                current_url = page.url
                print(f"Current URL: {current_url}")
                
                if "dashboard" in current_url or not page.is_visible('input[type="text"]'):
                    print("✅ Login successful!")
                else:
                    print("⚠️ Still on login page")
                    
            except Exception as e:
                print(f"Could not find login form: {e}")
                # Page might be loading or showing error
                page.screenshot(path="screenshot_error.png")
                print("Error screenshot saved")
                
        finally:
            browser.close()

if __name__ == "__main__":
    test_login()

File: test-login.js
const { chromium } = require('playwright');

(async () => {
  const browser = await chromium.launch({ headless: false });
  const page = await browser.newPage();

  try {
    console.log('Opening application...');
    await page.goto('http://localhost:5173');
    
    // Wait for login form to be visible
    await page.waitForSelector('input[type="text"]', { timeout: 5000 });
    
    // Take screenshot of login page
    await page.screenshot({ path: 'screenshot_login_page.png' });
    console.log('Login page loaded successfully');
    
    // Fill in credentials
    console.log('Entering credentials...');
    await page.fill('input[type="text"]', 'admin');
    await page.fill('input[type="password"]', 'password');
    
    // Take screenshot before clicking login
    await page.screenshot({ path: 'screenshot_before_login.png' });
    
    // Click login button
    console.log('Clicking login button...');
    await page.click('button[type="submit"]');
    
    // Wait for navigation or error message
    await page.waitForTimeout(2000);
    
    // Take screenshot after login attempt
    await page.screenshot({ path: 'screenshot_after_login.png' });
    
    // Check if we're still on login page (error) or navigated to dashboard
    const currentUrl = page.url();
    if (currentUrl.includes('dashboard') || !await page.isVisible('input[type="text"]')) {
      console.log('✅ Login successful! Redirected to:', currentUrl);
    } else {
      // Check for error message
      const errorText = await page.textContent('body');
      if (errorText.includes('失敗') || errorText.includes('error')) {
        console.log('❌ Login failed with error');
      } else {
        console.log('⚠️ Login status unclear');
      }
    }
    
  } catch (error) {
    console.error('Error during test:', error);
  } finally {
    await browser.close();
  }
})();